components/wifi_connector/wifi_ap.cpp

```diff
@@ -0,0 +1,72 @@
+#include "wifi_ap.hpp"
+#include "event_handlers.hpp"
+
+WifiAP::WifiAP()
+{
+  cfg_ap_ = {};
+  cfg_ap_.ap = {};
+  cfg_ap_.ap.ssid_hidden = 0;
+  cfg_ap_.ap.authmode = WIFI_AUTH_WPA_WPA2_PSK;
+
+  this->max_connections(4);
+
+  EventHandlers::add(WIFI_EVENT, this);
+}
+
+void WifiAP::ssid(std::string ssid)
+{
+  if (ssid.length() < 32) {
+    std::copy(ssid.begin(), ssid.end(), cfg_ap_.ap.ssid);
+    ssid_ = ssid;
+  }
+  else
+  {
+    ssid_ = "";
+  }
+}
+
+void WifiAP::password(std::string password)
+{
+  if (password.length() < 64) {
+    std::copy(password.begin(), password.end(), cfg_ap_.ap.password);
+  }
```

Comment:
If the SSID is too long, you set it to `""`, for the password there is no fallback behaviour, is this intended? What happens if the password is too long? Do we have a public (non-password-protected) AP then?